app/Models/EntityEvent.php

```diff
@@ -478,7 +484,9 @@ public function mostRecentOccurrence(int $year, int $month, int $day, array $mon
         if ($days > 1 && $this->length > 1) {
             $days -= $this->length - 1;
         }
-
+        if ($overDate) {
+            $days = -2;
```

Comment:
Why are we doing -2 here? It should be detailed in the code what's happening here, and why two.